File: frontend/src/types/availability.ts
export const WeekDayEnum = {
  Sunday: 0,
  Monday: 1,
  Tuesday: 2,
  Wednesday: 3,
  Thursday: 4,
  Friday: 5,
  Saturday: 6,
} as const;

export type WeekDayEnum = (typeof WeekDayEnum)[keyof typeof WeekDayEnum];

export interface Availability {
  dayOfWeek: WeekDayEnum;
  startTime: string; // "09:00"
  endTime: string; // "17:30"
  isActive: boolean;
}

File: backend/Models/Availability.cs
namespace backend.Models;

public class Availability
{
    public Guid Id { get; set; }

    public Guid MentorId { get; set; }

    /// <summary>
    /// 0 = Sunday ... 6 = Saturday
    /// </summary>
    public WeekDayEnum DayOfWeek { get; set; }

    public TimeSpan StartTime { get; set; }

    public TimeSpan EndTime { get; set; }

    public bool IsActive { get; set; }

    public DateTime CreatedAt { get; set; }

    public DateTime UpdatedAt { get; set; }

    public User Mentor { get; set; }
}
